pkg/services/govmomi/service.go

```diff
@@ -178,26 +179,26 @@ func (vms *VMService) ReconcileVM(ctx *context.VMContext) (vm infrav1.VirtualMac
 }
 
 // DestroyVM powers off and destroys a virtual machine.
-func (vms *VMService) DestroyVM(ctx *context.VMContext) (reconcile.Result, infrav1.VirtualMachine, error) {
+func (vms *VMService) DestroyVM(ctx context.Context, vmCtx *capvcontext.VMContext) (reconcile.Result, infrav1.VirtualMachine, error) {
 	vm := infrav1.VirtualMachine{
-		Name:  ctx.VSphereVM.Name,
+		Name:  vmCtx.VSphereVM.Name,
 		State: infrav1.VirtualMachineStatePending,
 	}
 
 	// If there is an in-flight task associated with this VM then do not
 	// reconcile the VM until the task is completed.
-	if inFlight, err := reconcileInFlightTask(ctx); err != nil || inFlight {
+	if inFlight, err := reconcileInFlightTask(vmCtx); err != nil || inFlight {
```

Comment:
```suggestion
	if inFlight, err := reconcileInFlightTask(ctx, vmCtx); err != nil || inFlight {
```
Need to refactor function reconcileInFlightTask